packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -192,6 +192,46 @@ export class SummarizerNode implements IRootSummarizerNode {
 		return { summary: result.summary, stats: result.stats };
 	}
 
+	public validateSummary() {
+		this.validateSummaryCore(false);
+	}
+
+	protected validateSummaryCore(parentSkipRecursion: boolean) {
+		assert(
+			this.wipSummaryLogger !== undefined,
+			0x1a3 /* "wipSummaryLogger should have been set in startSummary or ctor" */,
+		);
+		assert(this.wipReferenceSequenceNumber !== undefined, 0x1a4 /* "Not tracking a summary" */);
+		let didNodeSummarize = this.wipLocalPaths !== undefined;
+
+		if (parentSkipRecursion) {
+			if (this._latestSummary !== undefined) {
+				didNodeSummarize = true;
+			} else {
+				return;
```

Comment:
I see how this is parallel to `completeSummaryCore` but I'm curious if there's any additional validation to do in this case.

Reply:
yeah, it's the id of the uploaded summary from the server. It's not needed here. We can add it to `completeSummary` though.